presto-hive/src/main/java/com/facebook/presto/hive/metastore/BridgingHiveMetastore.java

```diff
@@ -177,6 +177,26 @@ public void renameColumn(String databaseName, String tableName, String oldColumn
         alterTable(databaseName, tableName, table);
     }
 
+    @Override
+    public void dropColumn(String databaseName, String tableName, String columnName)
+    {
+        Optional<org.apache.hadoop.hive.metastore.api.Table> source = delegate.getTable(databaseName, tableName);
+        if (!source.isPresent()) {
+            throw new TableNotFoundException(new SchemaTableName(databaseName, tableName));
+        }
+        org.apache.hadoop.hive.metastore.api.Table table = source.get();
+        if (table.getSd().getCols().size() == 1) {
+            throw new PrestoException(NOT_SUPPORTED, "Dropping last column is not supported");
```

Comment:
Let's put this user-facing validation to `HiveMetadata` so that it works for all metastore types. Though we could leave these here as safety checks, since this is a fundamentally racy operation. Is there some way to consolidate them?

How about making the message  "Cannot drop the only column in a table"

We also need validation that this is not a partition column.